#! activate environment
conda activate metagenome_env


# Index the reference genome (if not done before)
bwa index /path/to/host_reference_genome.fasta

# Align the paired-end reads to the reference genome using BWA MEM
bwa mem /path/to/host_reference_genome.fasta sample_reads_R1.fastq sample_reads_R2.fastq > aligned_reads.sam

# Convert SAM to BAM
samtools view -S -b aligned_reads.sam > aligned_reads.bam

# fetch host-unmapped reads

samtools fastq -@ 10 -f 12 -F 256 -1 sample_reads_R1.fastq.gz -2 sample_reads_R2.fastq.gz aligned_reads.bam &



# calculate the raw taxonomy profile for each sample
mkdir KRAKEN2/
kraken2 --db /path/to/ref_database/KRAKEN2_DB/ \
             --threads 16 --output KRAKEN2/sample.kraken --paired \
             --report KRAKEN2/sample.report sample_reads_R1.fastq.gz  sample_reads_R2.fastq.gz
# correct the raw profile generated by kraken2
bracken -d /path/to/ref_database/KRAKEN2_DB/ -i KRAKEN2/sample.report -o KRAKEN2/sample.bracken.txt" -w bracken/sample.kreport -l S -t 40

# transform .kreport file to .mpa
kreport2mpa.py --display-header -r KRAKEN2/sample.kreport -o KRAKEN2/sample.raw_count.mpa --read_count

# merge all sample profiles
combine_mpa.py -i KRAKEN2/*raw_count.mpa -o merged_kraken2.mpa

# remove temporary files
rm -rf *bam KRAKEN2/
